src/features/courses/course-controller.ts

```diff
@@ -72,11 +74,26 @@ class CourseController {
         });
     }
 
-    getTopicById(id: number): Promise<CourseTopicContent> {
+    getTopicById(id: number, userId: number | undefined = undefined): Promise<CourseTopicContent> {
```

Comment:
Typically if we have multiple parameters we use an object
Also instead of `userId: number | undefined = undefined` you should just do `userId?: number`

Reply:
Nit ^

Reply:
I chose not to use an object to minimize impact to other features. Optional parameters made more sense.

Interestingly, I can't initialize it to undefined with the questionmark. `Parameter cannot have question mark and initializer.`
I thought I tried the way you suggested before and it didn't work, but it seems like it is now. Thanks!

Reply:
yeah you can't do:
```
userId?: number = undefined
```
it has to be
```
userId?: number
```

Reply:
yeah for now I think it's fine, however we should try to stay consistent in the future, it's really my fault for not making it an object in the first place